packages/airnode-node/src/workers/cloud-platforms/aws.ts

```diff
@@ -1,49 +1,43 @@
 import { logger } from '@api3/airnode-utilities';
-import { goSync } from '@api3/promise-utils';
-import AWS from 'aws-sdk';
+import { go } from '@api3/promise-utils';
+import { LambdaClient, InvokeCommand } from '@aws-sdk/client-lambda';
 import { WorkerParameters, WorkerResponse } from '../../types';
 
-export function spawn(params: WorkerParameters): Promise<WorkerResponse> {
-  // lambda.invoke is synchronous so we need to wrap this in a promise
-  return new Promise((spawnResolve, spawnReject) => {
-    // Uses the current region by default
-    const lambda = new AWS.Lambda();
-
-    // AWS doesn't allow uppercase letters in lambda function names
-    const resolvedName = `airnode-${params.deploymentId}-run`;
-
-    const options = {
-      FunctionName: resolvedName,
-      Payload: JSON.stringify(params.payload),
-    };
-
-    const resolve: typeof spawnResolve = (value) => {
-      logger.debug(`Successful Lambda response: ${value}`);
-      spawnResolve(value);
-    };
-
-    const reject: typeof spawnReject = (error) => {
-      logger.debug(`Lambda invocation or execution failed. Response: ${error}`);
-      spawnReject(error);
-    };
-
-    // The Lambda invocation callback populates the error (first argument) only when the invocation fails (e.g. status
-    // code is 400) or the request is not parsed properly by the SDK and can't be invoked. For errors and timeouts that
-    // happen inside the invoked lambda have "data.FunctionError" and "data.Payload.errorMessage" populated instead.
-    // See: https://stackoverflow.com/q/42672023 and https://stackoverflow.com/q/48644093
-    lambda.invoke(options, (err, data) => {
-      if (err) return reject(err);
-
-      if (data.FunctionError) return reject(data.Payload);
-
-      if (!data.Payload) return reject(new Error('Missing payload in lambda response'));
-      const goPayload = goSync(() => JSON.parse(data.Payload!.toString()));
-      if (!goPayload.success) return reject(goPayload.error);
-
-      const goBody = goSync(() => JSON.parse(goPayload.data.body));
-      if (!goBody.success) return reject(goBody.error);
-
-      resolve(goBody.data as WorkerResponse);
-    });
-  });
+export async function spawn(params: WorkerParameters): Promise<WorkerResponse> {
+  // Uses the current region by default
+  const awsLambdaClient = new LambdaClient({});
+
+  // AWS doesn't allow uppercase letters in lambda function names
+  const resolvedName = `airnode-${params.deploymentId}-run`;
+
+  const options = {
+    FunctionName: resolvedName,
+    Payload: new TextEncoder().encode(JSON.stringify(params.payload)),
+  };
+
+  const invokeCommand = new InvokeCommand(options);
+  const goInvoke = await go(() => awsLambdaClient.send(invokeCommand));
+  if (!goInvoke.success) {
+    logger.debug(`Lambda invocation or execution failed. ${goInvoke.error}}`);
+    throw goInvoke.error;
+  }
+  const invokeOutput = goInvoke.data;
+
+  if (invokeOutput.FunctionError) {
```

Comment:
Where is this new API even documented? I tried quick googling but I only found https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html which is closest to "documentation" I'd expect, but it's still pretty useless.

Reply:
I know they probably haven't changed the lambda response, but still... Can you at least copy the comment that was there before?

Reply:
> Where is this new API even documented? I tried quick googling but I only found https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html which is closest to "documentation" I'd expect, but it's still pretty useless.

It is quite useless but that's the correct documentation.

> Can you at least copy the comment that was there before?

Will do.